### ISSUE TITLE
Start an ad network

### ISSUE THREAD

**Issue by chadwhitacre** [getsentry/osspledge.com#13]
Idea came up in a call with a popular dev portal. Basically:

- Make an API to serve random Pledgers.
- Ad companies consume API to promote OSS Pledge and Pledgers.

Bonus:

- If/when we implement #12, ad companies can report ad inventory as gift-in-kind contribution.

**Comment by chadwhitacre** [comment]
I talked to @stevekrouse about hosting this on Val.town, seems like it could be a good fit. :)

**Comment by stevekrouse** [comment]
Definitely!

**Comment by voxpelli** [comment]
Sounds like a good fit for https://www.ethicalads.io/ – some kind of partnering with them would be great as they were created to help the sustainability of OSS

Their background:

>  EthicalAds was built to sustain development of [Read the Docs](https://about.readthedocs.com/), an open source documentation platform. Advertising was the natural model because it scaled up well as our platform grew. However, we wanted to do ads right without tracking users or having security issues with third-party scripts.
> 
> Since no existing ad networks would partner with us on privacy-respecting ads, we built our own.

Their client script is MIT-licensed: https://github.com/readthedocs/ethical-ad-client

Their server is AGPL-licensed: https://github.com/readthedocs/ethical-ad-server

**Comment by chadwhitacre** [comment]
Ah, good call, @voxpelli! Waddya think @ericholscher? Worth a conversation? 😌

Ask is to donate some ad inventory to osspledge.com. 🙏

**Comment by vladh** [comment]
Suggestions from @gusaus:

* https://www.ethicalads.io/publishers/topics/open-source/
* https://opencollective.com/ethicalads

**Comment by vladh** [comment]
Thanks so much @ericholscher! It's great to see you here, especially since you [anticipated the Pledge][eric-blog] all the way back in 2018. ❤ 

I just found EthicalAds's [Community Ads program][community-ads] page.

@selviano, do you think it would be good for us to apply for an ad via this page, knowing that we can get back to Eric if we have any questions? Let me know if I can help with anything.

[eric-blog]: https://www.ericholscher.com/blog/2018/mar/9/one-percent-for-open-source/
[community-ads]: https://www.ethicalads.io/community-ads/

**Comment by selviano** [comment]
yeah it looks like we definitely qualify, I’ll fill out the form!On May 16, 2025, at 7:06 AM, Vlad-Stefan Harbuz ***@***.***> wrote:﻿
vladh left a comment (opensourcepledge/opensourcepledge.com#13)
Thanks so much @ericholscher! It's great to see you here, especially since you anticipated the Pledge all the way back in 2018. ❤
I just found EthicalAds's Community Ads program page.
@selviano, do you think it would be good for us to apply for an ad via this page, knowing that we can get back to Eric if we have any questions? Let me know if I can help with anything.

**Comment by selviano** [comment]
whoops, spoke too soon, I didn't realize that part of filling out the form was that you needed to provide ad creative and tagline etc. We should align on what we want the ads to say first. Let's discuss next steering committee meeting

**Comment by chadwhitacre** [comment]
Mockup:

<img width="209" height="282" alt="Image" src="https://github.com/user-attachments/assets/c0cf87d9-0093-4442-a714-e1327de20b7e" />

**Comment by chadwhitacre** [comment]
I think that's worth shipping. We make 31 more variants and write [a `pledge.py` importer](https://github.com/readthedocs/ethical-ad-server/tree/main/adserver/importers).

**Comment by chadwhitacre** [comment]
There's room for 20 characters in the image format. That works for everyone we have so far.

Here's the shortest:

<img width="205" height="294" alt="Image" src="https://github.com/user-attachments/assets/18b2d844-3fb2-493a-9d2f-187c647efcaa" />

**Comment by ericholscher** [comment]
I'd definitely simplify the image as much as possible. The display doesn't allow a ton of text, so I'd just include the brand logo and the OSS Pledge word mark at the bottom. Something like this: https://media.ethicalads.io/media/images/2020/01/house-hiredevs-02_junjKMN.png

I'd probably avoid the `Sucks` and keep the language clean, since you're running ads with people's brands attached, you're going to want them to look clean and agreeable.

If the brands won't change too much, we can probably handle the importer if you have a way of creating the images and giving us a zip or similar with the images.

**Comment by vladh** [comment]
<img width="240" height="180" alt="Image" src="https://github.com/user-attachments/assets/47c2143c-56a2-48c9-be79-82728d4feab4" />

**Comment by vladh** [comment]
98 chars:

> Fix burnout, stop vulns. Pay maintainers like Private Packagist does. Join the Open Source Pledge!

(“Private Packagist” is our longest member name.)

**Comment by chadwhitacre** [comment]
I agree in principle on KISS, but I don't like how the elephant seems to be labeled "Open Source Pledge."

**Comment by chadwhitacre** [comment]
**No burnout**, no vulns. Pay maintainers like Private Packagist does! Join the **Open Source Pledge**. 💃

**Comment by chadwhitacre** [comment]
?

![Image](https://github.com/user-attachments/assets/8e4582da-902b-4f71-96e6-181f05264f83)

**Comment by chadwhitacre** [comment]
> No burnout → no vulns.

Stretches the facts too much? No burnout → _fewer_ vulns?

**Comment by vladh** [comment]
All ad graphics done and ready for feedback, @chadwhitacre and @ericholscher.

https://drive.google.com/drive/folders/19M8ecyFK6gqRwJb7AMxrumxOObSeUIrl?usp=sharing
